Source/FieldSolver/FiniteDifferenceSolver/FiniteDifferenceAlgorithms/NodalAlgorithm.H

```diff
@@ -0,0 +1,134 @@
+        amrex::Gpu::ManagedVector<amrex::Real>& stencil_coefs_y,
+        amrex::Gpu::ManagedVector<amrex::Real>& stencil_coefs_z ) {
+
+        // Store the inverse cell size along each direction in the coefficients
+        stencil_coefs_x.resize(1);
+        stencil_coefs_x[0] = 1./cell_size[0];
+        stencil_coefs_y.resize(1);
+        stencil_coefs_y[0] = 1./cell_size[1];
+        stencil_coefs_z.resize(1);
+        stencil_coefs_z[0] = 1./cell_size[2];
+    }
+
+    /**
+    /* Perform derivative along x
+    /* (For a solver on a staggered grid, `UpwardDx` and `DownwardDx` take into
+    /* account the staggering; but for `NodalAlgorithm`, they are equivalent) */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    static amrex::Real UpwardDx(
+        amrex::Array4<amrex::Real> const& F,
+        amrex::Real const* coefs_x, int const n_coefs_x,
+        int const i, int const j, int const k ) {
+
+        amrex::Real const inv_dx = coefs_x[0];
+        return 0.5*inv_dx*( F(i+1,j,k) - F(i-1,j,k) );
+    };
+
+    /**
+    /* Perform derivative along x
+    /* (For a solver on a staggered grid, `UpwardDx` and `DownwardDx` take into
+    /* account the staggering; but for `NodalAlgorithm`, they are equivalent) */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    static amrex::Real DownwardDx(
+        amrex::Array4<amrex::Real> const& F,
+        amrex::Real const* coefs_x, int const n_coefs_x,
+        int const i, int const j, int const k ) {
+
+        amrex::Real const inv_dx = coefs_x[0];
+        return 0.5*inv_dx*( F(i+1,j,k) - F(i-1,j,k) );
+    };
+
+    /**
+    /* Perform derivative along y
+    /* (For a solver on a staggered grid, `UpwardDy` and `DownwardDy` take into
+    /* account the staggering; but for `NodalAlgorithm`, they are equivalent) */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    static amrex::Real UpwardDy(
+        amrex::Array4<amrex::Real> const& F,
+        amrex::Real const* coefs_y, int const n_coefs_y,
+        int const i, int const j, int const k ) {
+
+#if defined WARPX_DIM_3D
+            amrex::Real const inv_dy = coefs_y[0];
+            return 0.5*inv_dy*( F(i,j+1,k) - F(i,j-1,k) );
+#elif (defined WARPX_DIM_XZ)
+            return 0; // 2D Cartesian: derivative along y is 0
+#endif
+    };
+
+    /**
+    /* Perform derivative along y
+    /* (For a solver on a staggered grid, `UpwardDy` and `DownwardDy` take into
+    /* account the staggering; but for `NodalAlgorithm`, they are equivalent) */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    static amrex::Real DownwardDy(
+        amrex::Array4<amrex::Real> const& F,
+        amrex::Real const* coefs_y, int const n_coefs_y,
+        int const i, int const j, int const k ) {
+
+#if defined WARPX_DIM_3D
+            amrex::Real const inv_dy = coefs_y[0];
+            return 0.5*inv_dy*( F(i,j+1,k) - F(i,j-1,k) );
+#elif (defined WARPX_DIM_XZ)
+            return 0; // 2D Cartesian: derivative along y is 0
+#endif
```

Comment:
Should we replace these lines with
```c++
return UpwardDy(F, coefs_y, n_coefs_y, I, j, k);
```
? This can also be applied to other dimensions.